pkg/agentconfig/container.go

```diff
@@ -36,7 +36,9 @@ func (a *ContainerBuilder) AgentContainer(ctx context.Context) (*core.Container,
 			for _, ic := range PortUniqueIntercepts(cc) {
 				name := ic.ContainerPortName
 
-				if _, ok := names[name]; ok {
+				// We don't want to apply duplication logic to empty strings
```

Comment:
This seems a bit contrived. Why do a lookup if the name is "" and then check if it is ""?

I'd rather you kept the old implementation but just fence it with:
```
if name != "" {
  old impl here
}